aiida/cmdline/utils/migration/v03_to_v04.py

```diff
@@ -0,0 +1,528 @@
+            elif value['type'].startswith('node.process.'):
+                value['type'] = value['type'].replace('node.process.', 'process.', 1)
+
+
+def migration_rename_parameter_data_to_dict(data):
+    """Apply migration: 0029 - REV. 1.0.29
+    Update ParameterData to Dict
+    """
+    for value in data['export_data']['Node'].values():
+        if 'type' in value and value['type'] == 'data.parameter.ParameterData.':
+            value['type'] = 'data.dict.Dict.'
+
+
+def migration_dbnode_type_to_dbnode_node_type(metadata, data):
+    """Apply migration: 0030 - REV. 1.0.30
+    Renaming DbNode.type to DbNode.node_type
+    """
+    # For data.json
+    if 'Node' in data['export_data']:
+        for content in data['export_data']['Node'].values():
+            if 'type' in content:
+                content['node_type'] = content.pop('type')
+    # For metadata.json
+    if 'type' in metadata['all_fields_info']['Node']:
+        metadata['all_fields_info']['Node']['node_type'] = metadata['all_fields_info']['Node'].pop('type')
+
+
+def migration_remove_dbcomputer_enabled(metadata, data):
+    """Apply migration: 0031 - REV. 1.0.31
+    Remove DbComputer.enabled
+    """
+    # For data.json
+    if 'Computer' in data['export_data']:
+        for content in data['export_data']['Computer'].values():
+            if 'enabled' in content:
+                content.pop('enabled')
+    # For metadata.json
+    if 'enabled' in metadata['all_fields_info']['Computer']:
+        metadata['all_fields_info']['Computer'].pop('enabled')
+
+
+def migration_replace_text_field_with_json_field(data):
+    """Apply migration 0033 - REV. 1.0.33
+    Store dict-values as JSON serializable dicts instead of strings
+    NB! Specific for Django backend
+    """
+    import six
+    from aiida.common import json
+    if 'Computer' in data['export_data']:
+        for content in data['export_data']['Computer'].values():
+            for value in ['metadata', 'transport_params']:
+                if isinstance(content[value], six.text_type):
+                    content[value] = json.loads(content[value])
+    if 'Log' in data['export_data']:
+        for content in data['export_data']['Log'].values():
+            if isinstance(content['metadata'], six.text_type):
+                content['metadata'] = json.loads(content['metadata'])
+
+
+def migrate_v3_to_v4(metadata, data, folder, *args):  # pylint: disable=unused-argument
+    """
+    Migration of export files from v0.3 to v0.4
+
+    Note concerning migration 0032 - REV. 1.0.32:
+    Remove legacy workflow tables: DbWorkflow, DbWorkflowData, DbWorkflowStep
+    These were (according to Antimo Marrazzo) never exported.
+    """
+    old_version = '0.3'
+    new_version = '0.4'
+
+    verify_metadata_version(metadata, old_version)
+    update_metadata(metadata, new_version)
+
+    # Apply migrations on Nodes
+    if 'Node' in data['export_data']:
```

Comment:
This if else construction is a bit weird and difficult to understand. Please just call all functions and then make sure in the functions to properly check if the `Node` part is there.
Note you can easily often use `for value in data['export_data'].get('Node', {}).values()` to do the check inline so you don't have to have another `if 'Node' in data['export_data']` block

Reply:
done